src/components/MultiSelectGroup/MultiSelectGroup.test.tsx

```diff
@@ -0,0 +1,207 @@
+import * as React from "react";
+import renderer from "react-test-renderer";
+
+import MultiSelectGroup from "./MultiSelectGroup";
+import MultiSelect from "../MultiSelect/MultiSelect";
+
+const multiSelectItems = [
+  {
+    id: "colors",
+    name: "Colors",
+    items: [
+      { id: "red", name: "Red" },
+      { id: "blue", name: "Blue" },
+      { id: "yellow", name: "Yellow" },
+    ],
+  },
+  {
+    id: "pets",
+    name: "Pets",
+    items: [
+      { id: "cat", name: "Cat" },
+      {
+        id: "dog",
+        name: "Dog",
+        children: [
+          { id: "corgy", name: "Corgy" },
+          { id: "german-sheperd", name: "German Sheperd" },
+          { id: "afghan-hound", name: "Afghan Hound" },
+        ],
+      },
+      { id: "rat", name: "Rat" },
+    ],
+  },
+  {
+    id: "tools",
+    name: "Tools",
+    items: [
+      { id: "hammer", name: "Hammer" },
+      {
+        id: "skrewdriver",
+        name: "Skrewdriver",
+        children: [
+          { id: "slottet", name: "Slotted Skrewdriver" },
+          { id: "phillips", name: "Phillips Skrewdriver" },
+          { id: "allen", name: "Allen Skrewdriver" },
+        ],
+      },
+      { id: "whisk", name: "Whisk" },
+    ],
+  },
+];
+
+describe("MulitSelectGroup Accessibility", () => {
+  it("passes axe accessibility with string labels ", async () => {
+    const handleChangeMock = jest.fn();
+    const { container } = render(
+      <MultiSelectGroup
+        id="MultiSelectGroup"
+        labelText="MultiSelectGroup example"
+        showLabel={true}
+        multiSelectWidth="default"
+      >
+        {multiSelectItems.map((multiSelectItem) => (
+          <MultiSelect
+            key={multiSelectItem.id}
+            id={multiSelectItem.id}
+            variant={"listbox"}
+            label={multiSelectItem.name}
+            items={multiSelectItem.items}
+            selectedItems={{}}
+            onChange={handleChangeMock}
+            onClear={() => "clear"}
+          />
+        ))}
+      </MultiSelectGroup>
+    );
+    expect(await axe(container)).toHaveNoViolations();
+  });
+  it("renders with appropriate 'aria-label' attribute and value when 'showLabel' prop is set to false", () => {
+    const handleChangeMock = jest.fn();
+    const { rerender } = render(
+      <MultiSelectGroup
+        id="MultiSelectGroup"
+        labelText="MultiSelectGroup example"
+        showLabel={true}
+        multiSelectWidth="default"
+      >
+        {multiSelectItems.map((multiSelectItem) => (
+          <MultiSelect
+            key={multiSelectItem.id}
+            id={multiSelectItem.id}
+            variant={"listbox"}
+            label={multiSelectItem.name}
+            items={multiSelectItem.items}
+            selectedItems={{}}
+            onChange={handleChangeMock}
+            onClear={() => "clear"}
+          />
+        ))}
+      </MultiSelectGroup>
+    );
+    expect(screen.getByTestId("multi-select-group")).not.toHaveAttribute(
```

Comment:
I added the expect for the label text, however I noticed that the `<legend>` tag and text are still present on page even when hidden so it would pass even if the label is not visual on the page.

Reply:
That's correct. For the `fieldset` and `legend` structure, the `legend` must always be in the DOM, so what is done is to visually hide the text when `showLabel` is false. So this test is good as is now, thanks.